src/agent/TransportService.ts

```diff
@@ -0,0 +1,94 @@
+import { Logger } from '../logger'
+import { ConnectionRecord, ConnectionRole } from '../modules/connections'
+
+const DID_COMM_TRANSPORT_QUEUE = 'didcomm:transport/queue'
+
+export class TransportService {
+  private transportTable: TransportTable = {}
+  private logger: Logger
+
+  public constructor(logger: Logger) {
+    this.logger = logger
+  }
+
+  public saveTransport(connectionId: string, transport: Transport) {
+    this.transportTable[connectionId] = transport
+  }
+
+  public resolveTransport(connection: ConnectionRecord): Transport {
```

Comment:
Would be nice if we can make this more dynamic. this means we need to support the transport in core. Ideally adding an outbound/inbound transport class is enough

Reply:
Agree. I think we would need to move this logic to outbound transporters and just provide some utility functions to get the endpoint from DidDoc instead of having it in the core. That would also solve your other suggestions like "update this method to pass in supported outbound transport /schemes".